### PR TITLE
Refactor #9473: Move Vehicle Update into member function

### PR DESCRIPTION
**Pull request by dairemcdonald**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Don't close and open PRs unless you absolutely have to.

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -1920,141 +1897,140 @@ static uint16_t sub_6D7AC0(int32_t currentSoundId, int32_t currentVolume, int32_
...
 /**
  *
  *  rct2: 0x006D7BCC
  */
-static void vehicle_update_moving_to_end_of_station(rct_vehicle* vehicle)
+void rct_vehicle::UpdateMovingToEndOfStation()
 {
-    Ride* ride = get_ride(vehicle->ride);
-    int32_t flags, station;
+    Ride* newRide = get_ride(this->ride);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `this->`

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -2215,51 +2191,51 @@ static int ride_get_train_index_from_vehicle(Ride* ride, uint16_t spriteIndex)
...
  *  rct2: 0x006D7DA1
  */
-static void vehicle_update_waiting_for_passengers(rct_vehicle* vehicle)
+void rct_vehicle::UpdateWaitingForPassengers()
 {
-    vehicle->velocity = 0;
+    velocity = 0;
 
-    Ride* ride = get_ride(vehicle->ride);
+    Ride* newRide = get_ride(this->ride);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `this->`

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -2272,203 +2248,205 @@ static void vehicle_update_waiting_for_passengers(rct_vehicle* vehicle)
...
-    rct_ride_entry* rideEntry = get_ride_entry(vehicle->ride_subtype);
+    Ride* newRide = get_ride(this->ride);
+    rct_ride_entry* rideEntry = get_ride_entry(ride_subtype);
     if (rideEntry == nullptr)
     {
         return;
     }
-    rct_ride_entry_vehicle* vehicleEntry = &rideEntry->vehicles[vehicle->vehicle_type];
+
+    // rct_ride_entry_vehicle* vehicleEntry = &rideEntry->vehicles[vehicle_type];
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: This is required.

File: src/openrct2/ride/Vehicle.cpp [hunk 4]
```diff
@@ -2272,203 +2248,205 @@ static void vehicle_update_waiting_for_passengers(rct_vehicle* vehicle)
...
 
     // Mark the dodgem as in use.
-    if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_DODGEM_INUSE_LIGHTS && vehicle->animation_frame != 1)
+    if (flags & VEHICLE_ENTRY_FLAG_DODGEM_INUSE_LIGHTS && animation_frame != 1)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: You've accidently dropped the `vehicleEntry->`

File: src/openrct2/ride/Vehicle.cpp [hunk 5]
```diff
@@ -2525,149 +2503,148 @@ static void vehicle_update_waiting_to_depart(rct_vehicle* vehicle)
...
-    switch (ride->mode)
+    switch (newRide->mode)
     {
         case RIDE_MODE_BUMPERCAR:
             // Bumper mode uses sub_state / var_CE to tell how long
-            // the vehicle has been ridden.
-            vehicle->SetState(VEHICLE_STATUS_TRAVELLING_DODGEMS);
-            vehicle->var_CE = 0;
-            vehicle_update_dodgems_mode(vehicle);
+            // the this has been ridden.
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: vehicle

File: src/openrct2/ride/Vehicle.cpp [hunk 6]
```diff
@@ -3906,165 +3882,164 @@ static void vehicle_update_arriving(rct_vehicle* vehicle)
...
+        if (vehicle_open_restraints(this))
         {
-            vehicle->sub_state = 1;
+            sub_state = 1;
         }
     }
 
-    Ride* ride = get_ride(vehicle->ride);
-    if (ride->mode == RIDE_MODE_FORWARD_ROTATION || ride->mode == RIDE_MODE_BACKWARD_ROTATION)
+    Ride* newRide = get_ride(this->ride);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `this->`

File: src/openrct2/ride/Vehicle.cpp [hunk 7]
```diff
@@ -4077,143 +4052,144 @@ static void vehicle_update_unloading_passengers(rct_vehicle* vehicle)
...
 
 /**
  *
  *  rct2: 0x006D9CE9
  */
-static void vehicle_update_waiting_for_cable_lift(rct_vehicle* vehicle)
+void rct_vehicle::UpdateWaitingForCableLift()
 {
-    Ride* ride = get_ride(vehicle->ride);
+    Ride* newRide = get_ride(this->ride);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `this->`

File: src/openrct2/ride/Vehicle.cpp [hunk 8]
```diff
@@ -4636,15 +4612,15 @@ static bool vehicle_boat_is_location_accessible(const TileCoordsXYZ& location)
  *
  *  rct2: 0x006D9249
  */
-static void vehicle_update_swinging(rct_vehicle* vehicle)
+void rct_vehicle::UpdateSwinging()
 {
-    Ride* ride = get_ride(vehicle->ride);
-    rct_ride_entry* rideEntry = get_ride_entry(vehicle->ride_subtype);
+    Ride* newRide = get_ride(this->ride);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `this->`

File: src/openrct2/ride/Vehicle.cpp [hunk 9]
```diff
@@ -4636,15 +4612,15 @@ static bool vehicle_boat_is_location_accessible(const TileCoordsXYZ& location)
...
+    rct_ride_entry* rideEntry = get_ride_entry(ride_subtype);
 
-    // SubState for this ride means swinging state
+    // SubState for this Ride means swinging state
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: ride

File: src/openrct2/ride/Vehicle.cpp [hunk 10]
```diff
@@ -4653,180 +4629,180 @@ static void vehicle_update_swinging(rct_vehicle* vehicle)
...
  */
-static void vehicle_update_ferris_wheel_rotating(rct_vehicle* vehicle)
+void rct_vehicle::UpdateFerrisWheelRotating()
 {
     if (_vehicleBreakdown == 0)
         return;
 
-    Ride* ride = get_ride(vehicle->ride);
-    if ((vehicle->ferris_wheel_var_1 -= 1) != 0)
+    Ride* newRide = get_ride(this->ride);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `this->`

File: src/openrct2/ride/Vehicle.cpp [hunk 11]
```diff
@@ -4653,180 +4629,180 @@ static void vehicle_update_swinging(rct_vehicle* vehicle)
...
+            if (var_CE < newRide->rotations)
                 shouldStop = false;
         }
 
         if (shouldStop)
         {
-            ferris_wheel_var_0 = vehicle->ferris_wheel_var_0;
-            vehicle->ferris_wheel_var_0 = -abs(ferris_wheel_var_0);
-            vehicle->ferris_wheel_var_1 = abs(ferris_wheel_var_0);
+            ferris_wheel_var_0 = newFerrisWheelVar0;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: The other way round `newFerrisWheelVar0 = ferris_wheel_var_0`

File: src/openrct2/ride/Vehicle.cpp [hunk 12]
```diff
@@ -4653,180 +4629,180 @@ static void vehicle_update_swinging(rct_vehicle* vehicle)
...
+            newFerrisWheelVar0 = -abs(newFerrisWheelVar0);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `ferris_wheel_var_0 = -abs(newFerrisWheelVar0);`

File: src/openrct2/ride/Vehicle.cpp [hunk 13]
```diff
@@ -4653,180 +4629,180 @@ static void vehicle_update_swinging(rct_vehicle* vehicle)
...
+            ferris_wheel_var_1 = abs(newFerrisWheelVar0);
         }
     }
 
-    if (vehicle->ferris_wheel_var_0 != -8)
+    if (newFerrisWheelVar0 != -8)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: `ferris_wheel_var_0 `

**Comment by duncanspumpkin** [comment]
I'm closing this as it has gone too long without the changes being made. It will be just as easy to start again as it will be to make the changes now. In future it would have been much simpler to split this up into lots of smaller PR's that could have been commented individually and merged to prevent this large change.

**Comment by AntonySJohn** [comment]
Is the issue fixed? If not, I'd like to work on it.

**Comment by duncanspumpkin** [comment]
No one is working on it. If you do want to work on it please do just a few functions at a time and pr that.

**Pull request closed without merging**